File: stara/2006/python/zad6_c.py
# wczytanie slow
with open('../dane/dane.txt') as f:
    data = []
    for line in f.readlines():
        data.append(line[:-1])


# licznik palindromow
palis = 0
for word in data:
    # jezeli odrowcone jest takie samo to jest to palindrom
    if word[::-1] == word:
        palis += 1

# wyswietlenie odpowiedzi
answer = f'6 c) Ilosc palindromow: {palis}'
print(answer)

File: stara/2006/c++/zad6_c.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
using namespace std;

// funkcja sprawdzajaca czy dany string jest palindromem
bool isPali(string word)
{
    string reversed = "";

    for (int i = word.size() - 1; i >= 0; i--)
    {
        reversed += word[i];
    }

    if (word == reversed)
    {
        return true;
    }

    return false;
}

string zad6_c()
{
    string line;
    fstream file("../dane/dane.txt");
    vector<string> content;

    // wczytywanie danych z pliku
    if (file.is_open())
    {
        while (getline(file, line))
        {
            content.push_back(line);
        }
    }
    file.close();

    // sprawdzanie ile jest palindromow
    int palis = 0;

    for (int i = 0; i < content.size(); i++)
    {
        if (isPali(content[i]))
        {
            palis++;
        }
    }

    return "6 c) Liczba palindromow: " + to_string(palis);
}
